backend/de.metas.manufacturing/src/main/sql/postgresql/ddl/functions/PP_Product_BOM_Recursive.sql

```diff
@@ -1,105 +1,100 @@
 drop function if exists PP_Product_BOM_Recursive(numeric, varchar);
-create or replace function PP_Product_BOM_Recursive(p_PP_Product_BOM_ID numeric, p_ad_language varchar)
-returns table
-(
-	Line text,
-	Parent_Line text,
-	ProductValue varchar,
-	ProductName varchar,
-	QtyBOM numeric,
-	Percentage numeric,
-	UOMSymbol varchar,
-	--
-	depth integer,
-	M_Product_ID numeric,
-	IsQtyPercentage char(1),
-	C_UOM_ID numeric,
-    path integer[]
+CREATE OR REPLACE FUNCTION pp_product_bom_recursive(p_pp_product_bom_id numeric, p_ad_language character varying)
+    RETURNS TABLE(line text, parent_line text, productvalue character varying, productname character varying, qtybom numeric, percentage numeric, uomsymbol character varying, depth integer, m_product_id numeric, isqtypercentage character, c_uom_id numeric, path integer[], supplier text)
+    STABLE
+    LANGUAGE sql
+AS
+$$
+    --
+with recursive bomNode as (
+    (
+        select
+            array[1::integer] as path,
+            null::integer[] as parent_path,
+            1 as depth,
+            bomProduct.Value as ProductValue,
+            coalesce(pt.Name, bomProduct.Name) as ProductName,
+            bomProduct.M_Product_ID,
+            bomProduct.IsBOM,
+            bom.PP_Product_BOM_ID,
+            'N'::char(1) as IsQtyPercentage,
+            round(1::numeric, uom.StdPrecision) as QtyBOM,
+            null::numeric as Percentage,
+            COALESCE(uom.UOMSymbol, uomt.UOMSymbol) as UOMSymbol,
+            uom.C_UOM_ID,
+            COALESCE(bPartner.value, '') as supplier
```

Comment:
@mi-ps i am a bit hesitant to have it with `left joins` because in case there are multiple `C_BPartner_Product` records joined (not sure if possible, not sure which are the unique indexes) then all the data will be duplicated.

pls consider using having a sub-query, e.g. something like
```sql
(select bp.value 
from C_BPartner_Product bPartnerProduct
inner join C_BPartner bPartner on bPartnerProduct.c_bpartner_id = bPartner.c_bpartner_id
where 
bomProduct.m_product_id = bPartnerProduct.m_product_id and bPartnerProduct.iscurrentvendor='Y' and bPartnerProduct.isActive='Y'
-- and bPartner.isActive='Y' -- IMPORTANT show the bp value even if that's no longer active!
```

Reply:
@mi-ps (cc @teosarca ) there is 
```SQL
unique index uniquecurrentvendor
    on c_bpartner_product (iscurrentvendor, usedforvendor, m_product_id)
    where (iscurrentvendor = 'Y'::bpchar);
```

so we should not get multiple matches if the query is correct, which is not the case currently, as we are missing the `usedForVendor` check... even so, I think Teo's suggestion makes more sense, as that unique index might change in the future and we could get duplicate lines.. so, please do both :)